src/components/DownloadButton/DownloadButton.tsx

```diff
@@ -31,57 +30,55 @@ const DOWNLOAD_URLS = {
   }
 }
 
-export const DownloadButton: FC = memo(() => {
-  const [, advancedUserAgent] = useAdvancedUserAgentData()
-  const { track } = useAnalytics()
+export const DownloadButton: FC<{ osName: string | undefined; arch: string | undefined }> = memo(
```

Comment:
Update logic here to avoid asking twice for the osName and arch for the system and get it from the parent